code/__HELPERS/icons.dm

```diff
@@ -712,7 +712,8 @@ The _flatIcons list is a cache for generated icon files.
 			if(!current)
 				curIndex++ //Try the next layer
 				continue
-			currentLayer = current:layer
+			var/image/I = current

```

Comment:
Does it make much of a difference in `getFlatIcon` performance? It's not exactly a cheap proc.